packages/next-on-pages/src/buildApplication/buildMetadataFiles.ts

```diff
@@ -40,22 +40,36 @@ async function buildNextStaticHeaders(
 	);
 	const nextStaticHeaders = (nextStaticRoute as VercelSource)?.headers;
 
-	if (nextStaticHeaders) {
-		await writeFile(
-			join(outputDir, '_headers'),
-			`
+	if (!nextStaticHeaders) {
+		return;
+	}
+
+	const headersPath = join(outputDir, '_headers');
+
+	const nopContent = `
 # === START AUTOGENERATED @cloudflare/next-on-pages IMMUTABLE HEADERS ===
 ${nextStaticPath}/*
 ${Object.entries(nextStaticHeaders)
 	.map(([header, value]) => `  ${header}: ${value}`)
 	.join('\n')}
-# === END AUTOGENERATED @cloudflare/next-on-pages IMMUTABLE HEADERS ===\n`,
-			{
-				// in case someone configured redirects already, append to the end
-				flag: 'a',
-			},
-		);
+# === END AUTOGENERATED @cloudflare/next-on-pages IMMUTABLE HEADERS ===\n`;
+
+	try {
+		// Vercel hard links the public directory. We need to unlink before we
```

Comment:
Could you please include `https://github.com/vercel/vercel/blob/8e20fe/packages/gatsby-plugin-vercel-builder/src/helpers/static.ts#L15` in your comment so that we have a reference to the actual Vercel source code? 🙂🙏

Reply:
Good point. I ended up using a different code location that is more accurate. It's the same issue (static files being hard linked) but is not gatsby specific.

Reply:
ah great! 😄 

I didn't even notice the link was gatsby specific 🙈